sys/fs/devfs/random-vfs.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     sys_fs_devfs
+ * @{
+ *
+ * @file
+ * @brief       Random backends for devfs implementation
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include "vfs.h"
+
+#ifdef MODULE_PERIPH_HWRNG
+
+#include "periph/hwrng.h"
+
+static ssize_t hwrng_vfs_read(vfs_file_t *filp, void *dest, size_t nbytes);
+static int hwrng_vfs_open(vfs_file_t *filp, const char *name, int flags, mode_t mode, const char *abs_path);
+
+const vfs_file_ops_t hwrng_vfs_ops = {
+    .open = hwrng_vfs_open,
+    .read  = hwrng_vfs_read,
+};
+
+static int hwrng_vfs_open(vfs_file_t *filp, const char *name, int flags, mode_t mode, const char *abs_path)
+{
+    (void)filp;
+    (void)name;
+    (void)flags;
+    (void)mode;
+    (void)abs_path;
+
+    hwrng_init();
```

Comment:
is there any potential problem with reinitializing the hwrng on every open? could it mess up internal state on some platform?

Reply:
removed, as it's initialized from periph_init anyway